game.py

```diff
@@ -83,72 +87,79 @@ def main(self):
-            self.score()
-            self.movesmade()
+            STATS.score(self)
+            STATS.movesmade(self)
             pygame.display.update()
 
+    def select_square(self, x, y):
+        x, y = normalize_cords(x, y)
+        x_grid = int((x / 50) - 3)
+        y_grid = int((y / 50) - 3)
+        return x_grid, y_grid
+
+    def square_path(self, start, end):
+        path = astar(self.space, start, end)
+        first = path[0]
+        last = path[-1]
+        color = self.sqr_grid[first[0]][first[1]].color
+        self.space[last[0]][last[1]] = 1
+        self.space[first[0]][first[1]] = 0
+        self.move_square(path, color)
+        lines = self.find_adjacent_color(last, color)
+        self.check_length_remove_square(lines)
+        self.move_made = True
+        return path, color
+
     def handle_mouse_click(self):
         for event in pygame.event.get():
             if event.type == pygame.MOUSEBUTTONUP:
                 x, y = pygame.mouse.get_pos()
+                x_grid, y_grid = self.select_square(x, y)
                 if x < OFFSET or y < OFFSET:
                     break
-                x, y = normalize_cords(x, y)
-                x_grid = int((x / 50) - 3)
-                y_grid = int((y / 50) - 3)
                 if self.selected_square and self.space[x_grid][y_grid] == 0:
                     start = (
                         self.selected_square.grid_x,
                         self.selected_square.grid_y,
                     )
                     end = (x_grid, y_grid)
-                    path = astar(self.space, start, end)
+                    path = self.square_path(start, end)
                     if path is None:
                         break
-                    first = path[0]
-                    last = path[-1]
-                    color = self.sqr_grid[first[0]][first[1]].color
-                    self.space[last[0]][last[1]] = 1
-                    self.space[first[0]][first[1]] = 0
-                    self.move_square(path, color)
-                    lines = self.find_adjacent_color(last, color)
-                    # print(f'hor ---{lines[0]}    ver ---{lines[1]}')
-                    # print(f'UL-DR ---{lines[2]}     DL-UR ---{lines[3]}')
-                    self.check_length_remove_square(lines)
-                    self.move_made = True
                     self.selected_square = None
                     break
                 if self.space[x_grid][y_grid] == 0:
                     break
                 self.selected_square = self.sqr_grid[x_grid][y_grid]
             if event.type == pygame.QUIT:
-                self.save_score()
-                pygame.quit()
-                sys.exit()
+                self.buttons_instance.quit_menu(
+                    'Player name:', 'quit', 'reset', 'load',
+                    (255, 255, 255), 200, 200,
+                )
 
     def handle_selected_square(self):
         if self.selected_square:
```

Comment:
starting a function with an if is not a good practice? we just did a function call for no reason if the square is not selected